### PR TITLE
interfaces: interface hooks for refresh

### PR DESCRIPTION
**Pull request by stolowski**
This PR enables interface hooks on snap refresh, that is, hooks for existing connections will be re-executed when snap is updated. This is done by new "reconnect" task which injects interface hooks and connect tasks when updating the snap.

### CODE REVIEW

File: overlord/ifacestate/ifacestate.go [hunk 1]
```diff
@@ -107,14 +107,19 @@ func AutoConnect(st *state.State, change *state.Change, autoConnectTask *state.T
 	return connect(st, change, plugSnap, plugName, slotSnap, slotName, autoConnectTask)
 }
 
+// Reconnect returns a set of tasks for reconnecting an interface as part of snap installation.
+func Reconnect(st *state.State, change *state.Change, reconnectTask *state.Task, plugSnap, plugName, slotSnap, slotName string) (*state.TaskSet, error) {
+	return connect(st, change, plugSnap, plugName, slotSnap, slotName, reconnectTask)
```
Comment (niemeyer) [inline overlord/ifacestate/ifacestate.go]: Is there a benefit in having the exact same method with two different names and two different argument orders?

Comment (stolowski) [reply overlord/ifacestate/ifacestate.go]: Fair point. This is just to make the code more readable on the caller side and to emphasize that these are slightly different operations, although the difference is realized internally in `connect` based on properties of the autoconnectTask/reconnectTask.

I've adjusted the argument order of `connect` though so the order is same now.

Comment (niemeyer) [reply overlord/ifacestate/ifacestate.go]: They can't possibly be slightly different operations if they take exactly the same arguments and one of them just calls the other. There's no room for being different.

Comment (stolowski) [reply overlord/ifacestate/ifacestate.go]: I can reduce them to a single method if we agree on a name.. I couldn't find a good name and having `Reconnect` and `Autoconnect` helped avoid the naming problem :blush:

**Pull request merged**